cmd/argocd/commands/applicationset.go

```
@@ -0,0 +1,424 @@
+		Short: "Create one or more ApplicationSets",
+		Example: `
+	# Create ApplicationSets
+	argocd appset create <filename or URL> (<filename or URL>...)
+		`,
+		Run: func(c *cobra.Command, args []string) {
+			ctx := c.Context()
+
+			if len(args) == 0 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			argocdClient := headless.NewClientOrDie(clientOpts, c)
+			fileUrl := args[0]
+			appsets, err := cmdutil.ConstructApplicationSet(fileUrl)
+			errors.CheckError(err)
+
+			for _, appset := range appsets {
+				if appset.Name == "" {
+					err := fmt.Errorf("Error creating ApplicationSet %s. ApplicationSet does not have Name field set", appset)
+					errors.CheckError(err)
+					os.Exit(1)
+				}
+
+				conn, appIf := argocdClient.NewApplicationSetClientOrDie()
+				defer argoio.Close(conn)
+
+				// Get app before creating to see if it is being updated or no change
+				existing, err := appIf.Get(ctx, &applicationset.ApplicationSetQuery{Name: appset.Name})
+				if grpc.UnwrapGRPCStatus(err).Code() != codes.NotFound {
+					errors.CheckError(err)
+				}
+
+				appSetCreateRequest := applicationset.ApplicationSetCreateRequest{
+					Applicationset: appset,
+					Upsert:         upsert,
+				}
+				created, err := appIf.Create(ctx, &appSetCreateRequest)
+				errors.CheckError(err)
+
+				var action string
+				if existing == nil {
+					action = "created"
+				} else if !hasAppSetChanged(existing, created, upsert) {
+					action = "unchanged"
+				} else {
+					action = "updated"
+				}
+
+				fmt.Printf("application set '%s' %s\n", created.ObjectMeta.Name, action)
+			}
+		},
+	}
+	command.Flags().BoolVar(&upsert, "upsert", false, "Allows to override ApplicationSet with the same name even if supplied ApplicationSet spec is different from existing spec")
+	return command
+}
+
+// NewApplicationSetListCommand returns a new instance of an `argocd appset list` command
+func NewApplicationSetListCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
```

This seems to silently fail.

```
$ dist/argocd appset list
NAME  NAMESPACE  PROJECT  SYNCPOLICY  CONDITIONS  REPO  PATH  TARGET
$ kubectl config set-context --current --namespace argocd
Context "docker-desktop" modified.
$ dist/argocd appset list
NAME  NAMESPACE  PROJECT  SYNCPOLICY  CONDITIONS  REPO  PATH  TARGET
$ kubectl get appset
NAME        AGE
appset      35d
guestbook   118s
```